numpy/core/src/umath/ufunc_object.c

```diff
@@ -2868,32 +2868,42 @@ PyUFunc_Reduce(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *out,
     if (identity == NULL) {
         return NULL;
     }
+
+    /* Get the initializer */
+    if (initializer == NULL) {
+        initializer = identity;
+    }
+    else {
+        Py_DECREF(identity);
+        Py_INCREF(initializer);  /* match the reference count in the if above */
+    }
+
     /*
      * The identity for a dynamic dtype like
      * object arrays can't be used in general
      */
-    if (identity != Py_None && PyArray_ISOBJECT(arr) && PyArray_SIZE(arr) != 0) {
-        Py_DECREF(identity);
-        identity = Py_None;
-        Py_INCREF(identity);
+    if (initializer != Py_None && PyArray_ISOBJECT(arr) && PyArray_SIZE(arr) != 0) {
```

Comment:
Why do we do this? I think it is actually rather useful to be able to pass on something else than `None` (but can see that this would not be the case with a built-in `identity`)

Reply:
I'm leaving supporting object arrays to #8955, all the object-specific cases will be handled there.

Reply:
It's the same as the logic before, just on `initializer` instead of identity.

Reply:
OK, fair enough. Could you add a comment in the code still?

Reply:
One is already there. :-)